src/api/providers/kilocode.ts

```diff
@@ -59,253 +57,6 @@ export class KiloCodeHandler extends BaseProvider implements SingleCompletionHan
 	}
 }
 
-export class KiloCodeAnthropicHandler extends BaseProvider implements SingleCompletionHandler {
-	private options: ApiHandlerOptions
-	private client: Anthropic
-
-	constructor(options: ApiHandlerOptions) {
-		super()
-		this.options = options
-		const baseUri = getKiloBaseUri(options)
-		this.client = new Anthropic({
-			authToken: this.options.kilocodeToken,
-			baseURL: `${baseUri}/api/claude/`,
-			apiKey: null, //ignore anthropic apiKey, even if set in env vars - it's not valid for KiloCode anyhow
-		})
-	}
-
-	async *createMessage(systemPrompt: string, messages: Anthropic.Messages.MessageParam[]): ApiStream {
-		let stream: AnthropicStream<Anthropic.Messages.RawMessageStreamEvent>
-		const cacheControl: CacheControlEphemeral = { type: "ephemeral" }
-		let { id: modelId, maxTokens, thinking, temperature, virtualId } = this.getModel()
-
-		const userMsgIndices = messages.reduce(
-			(acc, msg, index) => (msg.role === "user" ? [...acc, index] : acc),
-			[] as number[],
-		)
-
-		const lastUserMsgIndex = userMsgIndices[userMsgIndices.length - 1] ?? -1
-		const secondLastMsgUserIndex = userMsgIndices[userMsgIndices.length - 2] ?? -1
-
-		try {
-			stream = await this.client.messages.create(
-				{
-					model: modelId,
-					max_tokens: maxTokens ?? ANTHROPIC_DEFAULT_MAX_TOKENS,
-					temperature,
-					thinking,
-					// Setting cache breakpoint for system prompt so new tasks can reuse it.
-					system: [{ text: systemPrompt, type: "text", cache_control: cacheControl }],
-					messages: messages.map((message, index) => {
-						if (index === lastUserMsgIndex || index === secondLastMsgUserIndex) {
-							return {
-								...message,
-								content:
-									typeof message.content === "string"
-										? [{ type: "text", text: message.content, cache_control: cacheControl }]
-										: message.content.map((content, contentIndex) =>
-												contentIndex === message.content.length - 1
-													? { ...content, cache_control: cacheControl }
```

Comment:
This stuff was slightly tricky, hence my question whether the openrouter implementation does this well too.